File: youtube-links/src/shared/features/playlists/commands/UpdatePlaylist.ts
import { ValidationConsts } from '@/shared/localization/ValidationConsts';

export namespace UpdatePlaylist {
  export interface Command {
    id: number;
    name: string;
    public: boolean;
  }

  export const Validation = {
    name: {
      notEmpty: (v: string) => !!v || 'Name should not be empty.',
      maximumLength: (v: string) =>
        v.length <= ValidationConsts.MaximumStringLength ||
        `The length of name must be ${ValidationConsts.MaximumStringLength} characters or fewer. You entered ${v.length} characters.`,
    },
  };
}

File: YoutubeLinks.Shared/Features/Playlists/Commands/UpdatePlaylist.cs
﻿using FluentValidation;
using MediatR;
using Microsoft.Extensions.Localization;
using YoutubeLinks.Shared.Localization;

namespace YoutubeLinks.Shared.Features.Playlists.Commands;

public static class UpdatePlaylist
{
    public class Command : IRequest<Unit>
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public bool Public { get; set; }
    }

    public class Validator : AbstractValidator<Command>
    {
        public Validator(IStringLocalizer<ValidationMessage> localizer)
        {
            RuleFor(x => x.Name)
                .NotEmpty()
                .WithMessage(x => localizer[nameof(ValidationMessageString.NameNotEmpty)])
                .MaximumLength(ValidationConsts.MaximumStringLength)
                .WithMessage(x => localizer[nameof(ValidationMessageString.NameMaximumLength)]);
        }
    }
}